proof/invariant-abstract/Invariants_AI.thy

```diff
@@ -693,10 +693,7 @@ definition
 where
   "valid_ep ep s \<equiv> case ep of
     IdleEP \<Rightarrow> True
-  | SendEP ts \<Rightarrow>
-      (ts \<noteq> [] \<and> (\<forall>t \<in> set ts. tcb_at t s) \<and> distinct ts)
-  | RecvEP ts \<Rightarrow>
-      (ts \<noteq> [] \<and> (\<forall>t \<in> set ts. tcb_at t s) \<and> distinct ts)"
+  | _ \<Rightarrow> (\<lambda>ts. ts \<noteq> [] \<and> (\<forall>t \<in> set ts. tcb_at t s) \<and> distinct ts) (ep_queue ep)"
```

Comment:
I'm almost tempted to use `let` here, but the lambda is actually quite nice.

Reply:
This is cool. I'm not sure what `ts` is supposed to abbreviate or remind me of. Would `q` or something make more sense? I'm not sure

Reply:
I suppose that’s meant to mean “(potentially) many `t`’s” just like using `xs` for a list...

Reply:
Ahh. I guess that makes some sense

Reply:
The general convention for lists is an `s` suffix for `sequence`, pronounced as plural (many `t`s as @mktnk3 says). So `ts` is a list/sequence of `t`.

Should create a TODO to add that to the conventions file, I thought it was in there already or part of the [general Isabelle naming conventions](https://isabelle.systems/conventions/naming.html), but couldn't find it. It's a tradition from SML and Haskell.